Library/Homebrew/compat/global.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Homebrew
+  class << self
+    module Compat
+      attr_writer :args
+
+      def args
+        unless @printed_args_warning
+          odeprecated "Homebrew.args", "`args = <command>_args.parse` and pass `args` along the call chain",
+                      disable_on: Time.new(2020, 12, 31)
```

Comment:
```suggestion
          odeprecated "Homebrew.args", "`args = <command>_args.parse` and pass `args` along the call chain"
```

Let's only `disable` this on a major release.

Reply:
I think it's better to use specific dates to give a better sense of urgency. Nobody knows when the next “major” version is released.

Reply:
In Homebrew/brew we've not been doing this and it's not part of our release process. You wanted the `date` here for cask stuff and I'm fine with that but I would like to avoid it here, please.

Reply:
> In Homebrew/brew we've not been doing this

Well, we should be doing this then. Our versioning is meaningless for deprecations anyways because there is no official way to downgrade to a previous one.

Reply:
> Well, we should be doing this then.

I disagree and I've been the person cutting most releases and writing the release notes. Regardless, this PR is not the best place for this discussion as there's no need to block this PR on convincing the rest of Homebrew to adopt a different deprecation method.

Reply:
@reitermarkus Latest on this?

Reply:
@reitermarkus Bump on this. Please either do this, escalate to the TSC or close out this PR and let me know and I'll do it, thanks!